applications/dashboard/models/class.rolemodel.php

```diff
@@ -750,4 +750,39 @@ public function deleteAndReplace($roleID, $newRoleID) {
         $result = $this->SQL->delete('Role', array('RoleID' => $roleID));
         return $result;
     }
+
+    /**
+     * Get a list of a user's roles that are permitted to be seen.
+     * Optionally return all the role data or just one field name.
+     *
+     * @param $userID
+     * @param string $field optionally the field name from the role table to return.
+     * @return array|null|void
+     */
+    public function getUsersViewableRoles($userID, $field = "Name") {
```

Comment:
maybe `getPublicUserRoles()` would be clearer? avoids User's/Users, uses UserRole (the name of the table), and Public is the inverse of Private/Personal.